File: THCS/2023_2024/CaMau_2324/Bai1/demkitu.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

void count_characters(const string& input_file, const string& output_file) {
    ifstream input(input_file);
    ofstream output(output_file);

    if (!input.is_open() || !output.is_open()) {
        cerr << "Cannot open file!" << endl;
        return;
    }

    string S;
    getline(input, S);  // Đọc xâu S

    vector<int> count(26, 0);  // Mảng đếm ký tự

    // Đếm số lần xuất hiện của các ký tự
    for (char charac : S) {
        count[charac - 'a']++;
    }

    // Ghi kết quả vào tệp tin
    for (int i = 0; i < 26; i++) {
        if (count[i] > 0) {
            output << char(i + 'a') << " " << count[i] << endl;  // Ghi ký tự và số lần xuất hiện
        }
    }

    input.close();
    output.close();
}

int main() {
    count_characters("demkitu.inp", "demkitu.out");
    return 0;
}

File: THCS/2023_2024/CaMau_2324/Bai1/demkitu.py
input_path = './demkitu.inp'
output_path = './demkitu.out'

def count_characters(input_file, output_file):
    # Đọc xâu từ tệp tin
    with open(input_file, 'r') as f:
        S = f.read().strip()  # Đọc xâu và loại bỏ ký tự trắng ở đầu/cuối

    # Mảng để đếm số lần xuất hiện của các ký tự
    count = [0] * 26  # 26 ký tự từ 'a' đến 'z'

    # Đếm số lần xuất hiện của mỗi ký tự
    for char in S:
        count[ord(char) - ord('a')] += 1

    # Ghi kết quả vào tệp tin
    with open(output_file, 'w') as f:
        for i in range(26):
            if count[i] > 0:  # Chỉ ghi các ký tự xuất hiện ít nhất 1 lần
                f.write(f"{chr(i + ord('a'))} {count[i]}\n")


count_characters(input_path, output_path)
